File: src/main/java/com/exsoloscript/mysteryteams/MysteryTeamsPlugin.java
package com.exsoloscript.mysteryteams;

import com.exsoloscript.mysteryteams.cmd.MysteryTeamsCommand;
import com.exsoloscript.mysteryteams.module.*;
import org.bukkit.ChatColor;
import org.bukkit.plugin.java.JavaPlugin;

public class MysteryTeamsPlugin extends JavaPlugin {

    private ModuleManager moduleManager;

    public void onEnable() {
        this.moduleManager = new ModuleManager(this);
        register();
    }

    public void onDisable() {

    }

    public void register() {
        getCommand("mt").setExecutor(new MysteryTeamsCommand(this));

        // Load modules

        this.getModuleManager().registerModule(new WoolModule());
        this.getModuleManager().registerModule(new BannerModule());
        this.getModuleManager().registerModule(new FireworkModule());
    }

    public ModuleManager getModuleManager() {
        return this.moduleManager;
    }

    public static String prefix() {
        return ChatColor.GOLD + "[" + ChatColor.RED + "MysteryTeams" + ChatColor.GOLD + "] " + ChatColor.WHITE;
    }
}

File: src/main/kotlin/com/exsoloscript/mysteryteams/MysteryTeamsPlugin.kt
package com.exsoloscript.mysteryteams

import com.exsoloscript.mysteryteams.cmd.MysteryTeamsCommand
import com.exsoloscript.mysteryteams.module.*
import org.bukkit.ChatColor
import org.bukkit.plugin.java.JavaPlugin

class MysteryTeamsPlugin : JavaPlugin() {

    val moduleManager: ModuleManager =  ModuleManager(this)

    override fun onEnable() {
        register()
    }

    fun register() {
        getCommand("mt").executor = MysteryTeamsCommand(this)

        // Load modules

        this.moduleManager.registerModule(WoolModule())
        this.moduleManager.registerModule(BannerModule())
        this.moduleManager.registerModule(FireworkModule())
    }

    companion object {

        fun prefix(): String {
            return "${ChatColor.GOLD}[${ChatColor.RED}MysteryTeams${ChatColor.GOLD}] ${ChatColor.WHITE}"
        }
    }
}
